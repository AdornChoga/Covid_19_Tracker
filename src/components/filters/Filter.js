import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { FaExclamationCircle } from 'react-icons/fa';
import { actionCreators } from '../../store/statistics/statistics';
import classes from './Filter.module.css';

const Filter = (props) => {
  const { fetchError } = props;
  const [error] = useState(fetchError);
  const [input, setInput] = useState('');
  const [dates, setDates] = useState({ currentDate: input, to: '', from: '' });
  const dispatch = useDispatch();

  useEffect(() => {
  }, [error]);

  const handleDateChange = (e) => {
    setInput(e.target.value);
    setDates({ ...dates, currentDate: e.target.value });
  };

  const handleSubmission = async (e) => {
    e.preventDefault();
    if (dates.currentDate !== '') {
      dispatch(actionCreators.filterByOneDate(dates.currentDate));
      dispatch(actionCreators.setAppStore(dates.currentDate, 1));
      if (!error) {
        dispatch(actionCreators.filtering());
      }
    }
  };

  const exitFilter = (e) => {
    if (e.target.id === 'exit_filter') {
      dispatch(actionCreators.filtering());
    }
  };

  return (
    <div className={classes.filter_container} id="exit_filter" onClick={exitFilter} aria-hidden="true">
      { error ? (
        <h5 className={classes.fetch_error_msg}>
          <FaExclamationCircle className={classes.info_icon} />
          Sorry, could not get data for that date
        </h5>
      )
        : ''}
      <form>
        <input type="date" value={input} className={classes.current_date} onChange={handleDateChange} />
        <br />
        <button type="submit" onClick={handleSubmission} className={classes.submit}>
          Get Statistics
        </button>
      </form>
    </div>
  );
};

Filter.propTypes = {
  fetchError: PropTypes.bool.isRequired,
};

export default Filter;
