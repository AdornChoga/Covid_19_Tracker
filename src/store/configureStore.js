import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import statistics from './statistics/statistics';

const reducer = combineReducers({
  statistics,
});

const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
