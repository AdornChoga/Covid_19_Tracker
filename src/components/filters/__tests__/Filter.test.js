import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../../store/configureStore';
import Filter from '../Filter.js';

it('should create filter component snapshot', () => {
  const fetchError = false;
  const tree = renderer.create(
  <Provider store={store}>
    <Router>
      <Filter fetchError={fetchError} />
    </Router>
  </Provider>);
  expect(tree).toMatchSnapshot();
})