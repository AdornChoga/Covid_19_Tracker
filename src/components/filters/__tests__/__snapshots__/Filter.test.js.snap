// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create filter component snapshot 1`] = `
<div
  aria-hidden="true"
  className="filter_container"
  id="exit_filter"
  onClick={[Function]}
>
  
  <form>
    <input
      className="current_date"
      onChange={[Function]}
      type="date"
      value=""
    />
    <br />
    <button
      className="submit"
      onClick={[Function]}
      type="submit"
    >
      Get Statistics
    </button>
  </form>
</div>
`;
